core/utils/poly-look/src/visualisations/charts/flow-charts/sankeyDiagram/sankeyDiagram.js

```diff
@@ -0,0 +1,203 @@
+ * @param {string} links[].target - Target of the link
+ * @param {number} links[].value - Value of the link
+ * @param {number = 400} width - The width of the svg
+ * @param {number = 300} height - The height of the svg
+ *
+ */
+export class SankeyDiagram extends Chart {
+  constructor({
+    selector,
+    links,
+    width = 400,
+    height = 300,
+    options,
+    margin,
+    color,
+  }) {
+    super({ selector, width, height });
+    this._links = links;
+    this._nodeAlign =
+      {
+        left: d3Sankey.sankeyLeft,
+        right: d3Sankey.sankeyRight,
+        center: d3Sankey.sankeyCenter,
+      }[options?.align] ?? d3Sankey.sankeyJustify;
+    this._margin = margin || defaultMargin;
+    this._nodeColor = color?.node || defaultNodeColor;
+    this._nodeLabelTextColor = color?.text || defaultTextColor;
+    this._nodeLabelTextOpacity =
+      color?.textOpacity || defaultNodeLabelTextOpacity;
+    this._nodeLabelBoxColor =
+      color?.nodeLabelBoxColor || defaultNodeLabelBoxColor;
+    this._nodeLabelBoxOpacity =
+      color?.nodeLabelBoxOpacity || defaultNodeLabelBoxOpacity;
+    this._linkColor = color?.link || defaultLinkColor;
+    this._linkOpacity = color?.linkOpacity || defaultLinkOpacity;
+  }
+
+  _createSVG() {
+    return d3
+      .select(this._selector)
+      .append("svg")
+      .attr("width", this._width)
+      .attr("height", this._height)
+      .attr("viewBox", [0, 0, this._width, this._height]);
+  }
+
+  _computeSankeyLayout(nodes, links) {
+    d3Sankey
+      .sankey()
+      .nodeId((d) => d.id)
+      .nodeAlign(this._nodeAlign)
+      .nodeWidth(nodeWidth)
+      .nodePadding(nodePadding)
+      .extent([
+        [this._margin.left, this._margin.top],
+        [this._width - this._margin.right, this._height - this._margin.bottom],
+      ])({ nodes, links });
+  }
+
+  _addNodes(nodesData) {
+    let nodes = this.chart.select(".nodes");
+    if (nodes.empty()) this.chart.append("g").attr("class", "nodes");
+    nodes
+      .selectAll("rect")
+      .data(nodesData)
+      .join("rect")
+      .attr("x", (d) => d.x0)
+      .attr("y", (d) => d.y0)
+      .attr("height", (d) => d.y1 - d.y0)
+      .attr("width", (d) => d.x1 - d.x0);
+  }
+
+  _addLinks(linksData) {
+    let links = this.chart.select(".links");
+    if (links.empty()) this.chart.append("g").attr("class", "links");
+    let pathGroups = links
+      .selectAll("g")
+      .attr("fill", "none")
+      .data(linksData)
+      .join("g")
+      .style("mix-blend-mode", linkMixBlendMode);
+
+    if (pathGroups.select("path").empty()) pathGroups.append("path");
+    pathGroups
+      .select("path")
+      .attr("d", linkPath)
+      .attr("stroke", this._linkColor)
+      .attr("stroke-width", ({ width }) => Math.max(1, width))
+      .attr("stroke-opacity", this._linkOpacity);
+  }
+
+  _addNodeLabelText(nodesData, nodeLabels) {
+    nodeLabels
+      .attr("font-family", "sans-serif")
+      .attr("font-size", 10)
```

Comment:
Sure about this font size? Again, possibly a constant?

Reply:
Will get to that, the text is not as it is supposed to be